queue_processors/queue_processor/queueproc_utils/tests/test_messaging_utils.py

```diff
@@ -9,19 +9,110 @@
+        mock_run.instrument.name = self.instrument_name
+        mock_run.experiment.reference_number = self.rb_number
+        mock_run.run_version = self.version
+        self.mock_run = mock_run
+
+    # pylint:disable=no-self-use
+    @patch(MESSAGE_CLASS_PATH + '._make_pending_msg')
+    @patch(MESSAGE_CLASS_PATH + '._send_pending_msg')
+    def test_send_pending(self, mock_send_pending, mock_make_pending):
+        """
+        Test: The correct control functions are called and the variables feed through as expected
+        When: Calling the control function send_pending
+        """
+        mock_make_pending.return_value = 'test'
+        MessagingUtils().send_pending('123')
+        mock_make_pending.assert_called_once_with('123')
+        mock_send_pending.assert_called_once_with('test', None)
+
+    @patch(MESSAGE_CLASS_PATH + '._make_pending_msg')
+    @patch(MESSAGE_CLASS_PATH + '._send_pending_msg')
+    def test_send_cancel(self, mock_send_pending, mock_make_pending):
+        """
+        Test: The correct control functions are called and the variables feed through as expected
+        When: Calling the control function cancel_pending
+        """
+        mock_msg = Mock()
+        mock_make_pending.return_value = mock_msg
+        MessagingUtils().send_cancel('123')
+        mock_make_pending.assert_called_once_with('123')
+        mock_send_pending.assert_called_once_with(mock_msg)
+        self.assertEqual(mock_msg.cancel, True)
+
+    @patch('model.database.access.start_database')
+    @patch(UTILS_PATH + '.reduction_run_utils.ReductionRunUtils.get_script_and_arguments')
+    def test_make_pending_msg(self, mock_get_scr_and_args, mock_start_db):
+        """
+        Test: A Message with expected value is generated
+        When: Calling the make_pending_msg
+        """
+        file_path = 'file/path'
+        script = 'script'
+        args = {'arg': 1}
+        # Setup Mock values
+        mock_get_scr_and_args.return_value = (script, args)
+
+        db_conn = Mock()
+        query_set = Mock()
+        mock_record = Mock()
+        mock_record.file_path = file_path
+        query_set.first.return_value = mock_record
+        db_conn.data_model.DataLocation.filter_by.return_value = query_set
+        mock_start_db.return_value = db_conn
+
+        actual = MessagingUtils()._make_pending_msg(self.mock_run)
+        self.assertIsInstance(actual, Message)
+        self.assertEqual(actual.run_number, self.run_number)
+        self.assertEqual(actual.instrument, self.instrument_name)
+        self.assertEqual(actual.rb_number, self.rb_number)
+        self.assertEqual(actual.data, file_path)
+        self.assertEqual(actual.reduction_script, script)
+        self.assertEqual(actual.reduction_arguments, args)
+        self.assertEqual(actual.run_version, self.version)
+        self.assertEqual(actual.facility, FACILITY)
+
+    @patch(UTILS_PATH + '.reduction_run_utils.ReductionRunUtils.get_script_and_arguments')
```

Comment:
NIT to be more python3 if it works still should we change this to an fstring?
`f"{UTILS_PATH}.reduction_run_utils.ReductionRunUtils.get_script_and_arguments"`